cranelift/jit/src/memory.rs

```diff
@@ -242,6 +268,51 @@ impl Memory {
         self.already_protected = self.allocations.len();
     }
 
+    /// Prepares an icache flush.
+    ///
+    /// [Memory::icache_flush] *must* be called after this.
```

Comment:
Nit - I think the phrasing of this comment gives the wrong impression because there is no obligation really to call `icache_flush()` at some point after this call; my suggestion is to just state that it is a prerequisite for any usage of the latter.